### PR TITLE
Ahub 19894 Validate requestTransformer if it is not null

### PR DESCRIPTION
**Pull request by marwamohamad**

### CODE REVIEW

**Comment by nhenneaux** [comment]
Could you rebase on master?

File: monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/SingleIpHttpClient.java [hunk 1]
```diff
@@ -117,10 +117,10 @@ void checkHealthStatus() {
                 httpRequestBuilder.timeout(Duration.ofMillis(serverConfiguration.getHealthReadTimeoutInMilliseconds()));
             }
 
-            Consumer<HttpRequest.Builder> requestTransformer = serverConfiguration.getRequestTransformer() != null ? serverConfiguration.getRequestTransformer() : builder -> {
-            };
-
-            requestTransformer.accept(httpRequestBuilder);
+            final Consumer<HttpRequest.Builder> requestTransformer = serverConfiguration.getRequestTransformer();
```
Comment (nhenneaux) [inline monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/SingleIpHttpClient.java:120]: You can set ServerConfiguration.DEFAULT_REQUEST_TRANSFORMER to null as well 
https://github.com/nhenneaux/resilient-httpclient/blob/master/single-host-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/singlehostclient/ServerConfiguration.java#L15

Comment (marwamohamad) [reply monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/SingleIpHttpClient.java:120]: So, that means in ah-core I will keep the HttpClientConfig for getHealthCheckRequestTransformer to return null?

Comment (nhenneaux) [reply monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/SingleIpHttpClient.java:120]: yes

**Pull request merged**